bridge/bridge/registry/mlflow.py

```diff
@@ -79,3 +81,78 @@ def fetch_version_artifact(
             raise Exception(
                 "Could not determine model root path. No MLmodel definition."
             )
+
+    def tag_deployed_models(
+        self, deployed_models: List[Model], registered_models: List[Model]
+    ):
+        all_model_names = set[str]()
+        registered_versions = set[ModelVersion]()
+        for model in registered_models:
+            for versions in model.versions.values():
+                for mv in versions:
+                    all_model_names.add(mv.model_name)
+                    registered_versions.add(mv)
+
+        deployed_versions = set[ModelVersion]()
```

Comment:
I think there is a problem with this set based approach. If a model is moved between stages (from Staging -> Production), then it will be deploying for several minutes, but this will not be visible here. 

I think we need to compare the set of models within each stage, to know if it is truly deployed and up to date, or not. 

@JoshBroomberg what will the `deployed_models` returned from the Sagemaker deploy target look like in this scenario? Will the version be absent from both stages?

Reply:
Agree